generator/src/Spec/HaskellLexicalSyntax.hs

```diff
@@ -78,6 +81,9 @@ specialid = as"as" ! as"qualified" ! as"hiding"
 varsym = (a symbol & many (a symbol ! aa ":")) -! (reservedop ! dashes)
 consym = (aa ":" & many (a symbol ! aa ":")) -! reservedop
 
+quasiquote = aa "[" & varid & aa "|" & many (a cany ! whitechar) & aa "|]"
+qquasiquote = aa "[" & qvarid & aa "|" & many (a cany ! whitechar) & aa "|]"
+
```

Comment:
I don't think that's quite right as we want the `|]` to terminate the quote.  With this regular expression multiple quoters will get combined into a single quoter, I think.  For example, consider  `[|hello|world|] [|this|test|]`, if I understand correctly, we'll match this as a single quoter all the way to the `test`.

On the topic of separating the quoter from the quoter body we could generate two tokens for each quoter: `startQuasiQuoter`, which would match the `[|name|` part, and `quoterBody`, which would match the rest.

Reply:
Thanks I'll do that!

Reply:
> I don't think that's quite right as we want the `|]`

Am I right in assuming that this will need to be handled by an external function like with nested comments? I can't see how to do it with the regex options available.

Reply:
I tried `many ((a cany ! whitechar) -! aa "|]")` but it seems to terminate on the first `|` it finds.

Reply:
I just spent some time mucking around with this, and I think I came up with a version that works---I pushed it to `master`.  Would you mind giving it a go?

Reply:
It didn't seem to work for me:
```
> ghci> lexerPass0 "[name|body|]"
[(Special,(Pos {char = 0, line = 1, column = 1},"[")),(Varid,(Pos {char = 1, line = 1, column =
 2},"name")),(Reservedop,(Pos {char = 5, line = 1, column = 6},"|")),(Varid,(Pos {char = 6, lin
e = 1, column = 7},"body")),(Reservedop,(Pos {char = 10, line = 1, column = 11},"|")),(Special,
(Pos {char = 11, line = 1, column = 12},"]"))]
```

I do have something working in the same style as `NestedCommentStart` if that would be acceptable?

Reply:
Oh I see, it's just slightly incorrect syntax actually

Reply:
If I change this it works:
```diff
-  start      = as "[|" & (qvarid ! varid) & as "|"
+  start      = as "[" & (qvarid ! varid) & as "|"
```

Reply:
Ah yes, it's been a while since I'd used a quasi quote so I'd misremembered the syntax.  Should be fixed now.

Reply:
Yes it works perfectly now for the cases I have tried!

I would like to add this to `pretty-show` now, do you have a preference on constructor name? `QQuote` could be used but imports would need to be qualified in the parser.

The change is pretty simple, maybe you even just prefer to do it yourself?

Reply:
I do have a PR up with just `Quote` but happy to change it to whatever